### PR TITLE
Use vcpkg for dependencies

### PR DESCRIPTION
**Pull request by DethRaid**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -255,21 +256,24 @@ add_coverage(nova-renderer)
...
-                     spirv-cross-hlsl
                      spirv-cross-reflect
+                     spirv-cross-msl
                      spirv-cross-util
-                     spirv::headers
-                     vma::vma
+                     unofficial::vulkan-memory-allocator::vulkan-memory-allocator
+                     Vulkan::Vulkan
                      vulkan::sdk
                      ${Vulkan_LIBRARIES})
```
Comment (cwfitzgerald) [inline CMakeLists.txt]: Doesn't need to be here if Vulkan::Vulkan is there.

File: readme.md [hunk 2]
```diff
@@ -17,6 +17,16 @@ The Nova Renderer is currently early in development, so early that I don't think
 
 If you want to see what we have planned, or what has already been done, check out [the GitHub projects page](https://github.com/NovaMods/nova-renderer/projects).
 
+## Developer setup
+
+Nova uses [vcpkg](https://github.com/Microsoft/vcpkg) for dependency management. You need to install it on your system PATH. Once `vcpkg` is on your path, run the appropriate setup script for your platform - run `setup-windows.ps1` from Powershell on Windows, run `setup-linux.sh` from Bash on Linux. This will install almost all of Nova's dependencies
+
+Install the LunarG Vulkan SDK. Use the latest SDK for Vulkan 1.1
+
+Generate your compiler's build files from the CMake project. We recommend creating your build files in a `build` directory in the root of your local Nova Renderer repository. Be sure to pass CMake the `vcpkg` toolchain file, or your CMake will be very sad
```
Comment (cwfitzgerald) [inline readme.md]: This could use some examples of how to do that.

File: setup-linux.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+vcpkg install miniz:x64-linux fmt:x64-linux glm:x64-linux glslang:x64-linux gtest:x64-linux nlohmann-json:x64-linux spirv-cross:x64-linux spirv-tools:x64-linux vulkan-memory-allocator:x64-linux vulkan:x64-linux spirv-tools:x64-linux minitrace:x64-linux
```
Comment (cwfitzgerald) [inline setup-linux.sh]: End of line

File: setup-windows.ps1 [hunk 4]
```diff
@@ -0,0 +1 @@
+vcpkg install miniz:x64-windows fmt:x64-windows glm:x64-windows glslang:x64-windows gtest:x64-windows nlohmann-json:x64-windows spirv-cross:x64-windows spirv-tools:x64-windows vulkan-memory-allocator:x64-windows vulkan:x64-windows spirv-tools:x64-windows minitrace:x64-windows
```
Comment (cwfitzgerald) [inline setup-windows.ps1]: End of line

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
Almost done, I think we just need to unfuck appveyor and jenkins once the final PR to vcpkg gets cleared

**Review by cwfitzgerald — DISMISSED** [review]

**Comment by DethRaid** [comment]
CI can't build this PR until https://github.com/Microsoft/vcpkg/pull/5577 is merged

**Comment by DethRaid** [comment]
test this please

**Comment by DethRaid** [comment]
test this please

**Comment by cwfitzgerald** [comment]
Merging blocked until https://github.com/Microsoft/vcpkg/pull/5699 merged.

**Review by cwfitzgerald — APPROVED** [review]

**Pull request merged**